support/cas-server-support-pac4j-core-clients/src/main/java/org/apereo/cas/support/pac4j/web/flow/LimitedTerminateSessionAction.java

```diff
@@ -0,0 +1,89 @@
+package org.apereo.cas.support.pac4j.web.flow;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.apereo.cas.util.Pac4jUtils;
+import org.apereo.cas.web.support.WebUtils;
+import org.pac4j.core.profile.ProfileManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.webflow.action.AbstractAction;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+import com.google.common.base.Throwables;
+
+
+/**
+ * A Spring Flow action that invalidates user session.
+ * 
+ * Intended to be used at the end of the flow, in the "on-end" section.
+ * 
+ * Sometimes we need the session until the very last moment of the flow (e.g. during logout, we need the session to render the SAML SLO
+ * request), that's why we invalidate it after the flow terminates.
+ * 
+ * It contains a subset of code from the original {@code TerminateSessionAction} that is not executed on deferred session termination.
+ * So these two actions are complementary:
+ * <ul>
+ * <li>Either only {@code TerminateSessionAction} is used,</li>
+ * <li>or both are used, {@code TerminateSessionAction} does everything except actual HTTP session termination and PAC4J logout, which is
+ * done in {@code LimitedTerminateSessionAction}.</li>
+ * </ul>
+ * 
+ * @author jkacer
+ * 
+ * @since 5.3.0
+ */
+public class LimitedTerminateSessionAction extends AbstractAction {
```

Comment:
This needs go away.

Reply:
Is there a reason why it can't stay? I use it in the case when we cannot invalidate the HTTP session yet during logout because of PAC4J Manager, so I postpone the invalidation. And it's only used with PAC4J here: Pac4jWebflowConfigurer#doInitialize(). It's not meant as a duplicate of TerminateSessionAction.

Reply:
A possible solution would be to move the normal TerminateSessionAction to the end of the flow in Pac4jWebflowConfigurer. I would not need this action then. Not sure what is better / worse.